src/components/shared/PinchZoomView.tsx

```diff
@@ -0,0 +1,100 @@
+import { ImageProps, SafeAreaView, TouchableOpacity, View } from 'react-native';
+import ImageViewer, { ImageViewerPropsDefine } from 'react-native-image-zoom-viewer';
+import React, { ReactElement } from 'react';
+import { Ionicons } from '@expo/vector-icons';
+import styled from 'styled-components/native';
+
+export interface ImageI extends HeaderProps {
+  url: string;
+  width?: number;
+  height?: number;
+  props?: ImageProps;
+}
+
+interface Props extends Omit<ImageViewerPropsDefine, 'imageUrls'> {
+  images: ImageI[];
+  onPressClose?: () => void;
+}
+
+const HeaderContainer = styled.SafeAreaView`
+  position: absolute;
+  width: 100%;
+  z-index: 1;  
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+`;
+
+const Title = styled.Text`
+  color: white;
+  font-size: 20;
+`;
+
+const SubTitle = styled.Text`
+  color: white;
+  font-size: 16;
+`;
+
+interface HeaderProps {
+  title?: string;
+  subtitle?: string;
+  onPressClose?: () => void;
+}
+
+const Header = ({
+  title, subtitle, onPressClose,
+}: HeaderProps): React.ReactElement<HeaderProps, typeof SafeAreaView> => {
```

Comment:
For return type, `React.ReactElement` would be enough. It currently looks redundant.